### PR TITLE
EOS-19686: Logic incorporated for unsupported feature

### PR DESCRIPTION
**Pull request by ypise**
Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

### CODE REVIEW

File: srv/_modules/cortx_unsupported_features.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# How to test:
+# $ salt-call saltutil.clear_cache
+# $ salt-call saltutil.sync_modules
+
```
Comment (83bhp) [inline srv/_modules/cortx_unsupported_features.py:21]: Have you missed to add the actual test command here?

File: srv/components/csm/config/config.sls [hunk 2]
```diff
@@ -15,6 +15,19 @@
...
 
+
+Register features to be unsuported:
+  module.run:
+    - cortx_unsupported_features.register:
+      - component: provisioner
+      - unsupported_feature_list:
+        - fw_update
+      - setup_types:
+        - VM
```
Comment (83bhp) [inline srv/components/csm/config/config.sls:26]: Since the setup type RBOD is missing here, does that mean the fw update will be supported only for RBOD?

**Review by 83bhp — APPROVED** [review]
minor comment/question for understanding, patch looks ok. Thanks!

File: srv/components/csm/config/config.sls (hunk 2 above)
Comment (ypise) [reply srv/components/csm/config/config.sls:26]: Yes for now we should maintain support for RBOD. But once it is officially dropped from our supported features list, we can disable it for RBOD as well by listing here.

File: srv/_modules/cortx_unsupported_features.py (hunk 1 above)
Comment (ypise) [reply srv/_modules/cortx_unsupported_features.py:21]: Actual test command is tricky. However, I'll include the test process here.

File: srv/components/csm/config/config.sls (hunk 2 above)
Comment (ypise) [reply srv/components/csm/config/config.sls:26]: I have made the addition to `test/r2.md` for manual testing.

File: srv/_modules/cortx_unsupported_features.py (hunk 1 above)
Comment (ypise) [reply srv/_modules/cortx_unsupported_features.py:23]: Updated the basic commands necessary for testing.

Comment (83bhp) [reply srv/_modules/cortx_unsupported_features.py:23]: Thanks!! 
There are some codacy warnings, please address.

**Review by 83bhp — APPROVED** [review]

**Pull request merged**